packages/camera/camera_web/lib/src/camera.dart

```diff
@@ -171,4 +174,69 @@ class Camera {
       ..objectFit = 'cover'
       ..transform = 'scaleX(-1)';
   }
+
+  html.MediaRecorder? _mediaRecorder;
+  final StreamController<VideoRecordedEvent> _videoRecorderController =
+      StreamController();
+
+  /// Returns a Stream that emits when a video Recodring with a defined maxVideoDuration was created
+  Stream<VideoRecordedEvent> get onVideoRecordedEventStream =>
+      _videoRecorderController.stream;
+
+  /// Starts a new Video Recording using [html.MediaRecorder]
+  /// /// Throws a [html.DomException.INVALID_STATE] if there already is an active Recording
+  Future<void> startVideoRecording({Duration? maxVideoDuration}) async {
+    if (_mediaRecorder != null && _mediaRecorder!.state != 'inactive') {
+      throw html.DomException.INVALID_STATE;
+    }
+    _mediaRecorder ??= html.MediaRecorder(
+        videoElement.captureStream(), {'mimeType': 'video/webm'});
+
+    if (maxVideoDuration != null) {
+      _mediaRecorder!.addEventListener('dataavailable', (event) {
+        final blob = (event as html.BlobEvent).data;
+        final file = XFile(html.Url.createObjectUrl(blob));
+        _videoRecorderController
+            .add(VideoRecordedEvent(this.textureId, file, maxVideoDuration));
+        _mediaRecorder!.stop();
+      });
+      _mediaRecorder!.start(maxVideoDuration.inMilliseconds);
+    } else {
+      _mediaRecorder!.start();
+    }
+  }
+
+  /// Pauses the current video Recording
+  /// Throws a [html.DomException.INVALID_STATE] if there is no active Recording
+  Future<void> pauseVideoRecording() async {
+    if (_mediaRecorder == null || _mediaRecorder!.state == 'inactive') {
+      throw html.DomException.INVALID_STATE;
+    }
+    _mediaRecorder?.pause();
+  }
+
+  /// Resumes a video Recording
+  /// Throws a [html.DomException.INVALID_STATE] if there is no active Recording
+  Future<void> resumeVideoRecording() async {
+    if (_mediaRecorder == null || _mediaRecorder!.state == 'inactive') {
+      throw html.DomException.INVALID_STATE;
+    }
```

Comment:
(These exceptions are already thrown by the browser, no need to assert for _mediaRecorder.state)

Reply:
See above